interface/src/app/sample/services.js

```diff
@@ -0,0 +1,37 @@
+angular.module('adage.sample.services', [
+  'ngResource',
+  'adage.utils'
+])
+
+.factory('Sample', ['$resource', '$http', 'ApiBasePath',
+  function($resource, $http, ApiBasePath) {
+    var Sample = $resource(
+      ApiBasePath + 'sample/:id/',
+      // TODO need to add logic for handling pagination of results.
+      // then, can change "limit" below to something sensible
+      {id: '@id', limit: 0},
+      // Angular expects a query service to give only a list but our Tastypie
+      // interface wraps the response list with pagination so we tell Angular to
+      // expect an object instead via isArray: false
+      {
+        'get': {
+          method: 'GET',
+          isArray: false
+        },
+        'getExperiments': {
+          url: ApiBasePath + 'sample/:id/get_experiments/',
+          method: 'GET',
+          isArray: true
+        }
+      }
+    );
+    Sample.getUri = function(uri) {
+      // We can't easily get Tastypie-supplied resource_uris via $resource, so
+      // this uses $http directly and returns the HttpPromise
+      return $http({url: uri, method: 'GET'});
+    };
+    return Sample;
+  }
+])
+
```

Comment:
Right. Just to be clear: I did actually try the change to `$resource` you suggested and got a couple of stack traces back from the tests so backed it out.

I completely agree with what you've both noticed about naming as well. If this function even survives reorganization (might not be necessary anymore with a good refactoring) it should definitely have a better name.